lsp_types_legacy.py

```diff
@@ -0,0 +1,5892 @@
+# Code generated. DO NOT EDIT.
+# LSP v3.17.0
+
+from typing_extensions import NotRequired
```

Comment:
typing_extensions requires python 3.7 and up

Reply:
but this code is meant to support python 3.3

Reply:
As I've mentioned in the readme, those are special types to be used only by ST LSP with custom typings. We would not use those imports in ST.

So the options are:
 - remove those import from *legacy variant
 - treat *legacy file as still only compatible with 3.7 but using alternative syntax. Though of course there is little sense in anyone using those outside of ST anyway.

Reply:
Through removing the imports would make the files not validate using pyright/mypy anymore so I'd rather keep those.

Reply:
Through removing the imports would make the files not validate using pyright/mypy anymore so I'd rather keep those.

Reply:
I am ok with removing those imports.
I am also ok to use Optional instead of NotRequired for legacy?

Reply:
a) We can no longer validate with pyright when imports are removed so I don't think it's a good idea
b) I might have to explain it more because I see some confusion:
- The *legacy types strive to avoid using syntax that is not compatible with Python 3.3 so that we can use those in ST.
- Those need to not crash when running in 3.3 and that's the point of alternative version
- We still want to take advantage of `NotRequired` when developing LSP locally so using `Optional` instead would be missing the point
- LSP does this versions-based logic here: https://github.com/sublimelsp/LSP/blob/b84f25ec917196d860032bea0d5cb20942f13e54/plugin/core/typing.py#L3-L33 so that at runtime we are using dummy type aliases that do nothing but when type-checking with pyright or mypy an actuall `typings` and `typing_extensions` are used and type checking works correctly, including `NotRequired`

Reply:
Of course there is probably some room for improvement in regards of documenting what it is exactly and potentially improving my comment in the readme file.

Reply:
Do you think that moving the legacy lsp types to a folder named sublime would help avoid confusion?

Reply:
maybe just call them `*_sublime_text.py`?

Reply:
once that commit is pushed, I will rebase this.